File: app/src/main/java/com/example/annahockeyleague/Adapters/TopScorersAdapter.java
package com.example.annahockeyleague.Adapters;

import android.content.Context;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.example.annahockeyleague.Entity.TopScorers;
import com.example.annahockeyleague.R;
import com.squareup.picasso.Picasso;


import java.util.ArrayList;

public class TopScorersAdapter extends ArrayAdapter<TopScorers> {

    private static final String TAG = "TopScorerAdapter";

    public TopScorersAdapter(@NonNull Context context, @NonNull ArrayList<TopScorers> objects) {
        super(context, 0, objects);
        Log.d(TAG,"top scorer adapter constructor");
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {

        Log.d(TAG,"method get view");

        if(convertView == null)
        {
            Log.d(TAG, "convertview is null");

            convertView = LayoutInflater.from(getContext()).inflate(R.layout.top_scorer_row_template,parent,false);
        }

        TopScorers scorersData = getItem(position);

        ((TextView) convertView.findViewById(R.id.top_scorer_name)).setText(scorersData.getPlayer().getName());
//        ((TextView) convertView.findViewById(R.id.top_scorer_team)).setText(scorersData.getTeamName());
        ((TextView) convertView.findViewById(R.id.top_scorer_goals)).setText(String.valueOf(scorersData.getGoals()));

        Picasso.get().load(R.drawable.sample_kumble).into((ImageView) convertView.findViewById(R.id.top_scorer_avatar));

        return convertView;

    }
}

File: app/src/main/java/com/ahl/annahockeyleague/kotlin/adapters/TopScorersAdapter.kt
package com.ahl.annahockeyleague.kotlin.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.ahl.annahockeyleague.R
import com.ahl.annahockeyleague.kotlin.data.TopScorers
import kotlinx.android.synthetic.main.top_scorer_row_template.view.*

class TopScorersAdapter : RecyclerView.Adapter<TopScorersAdapter.TopScorersViewHolder>() {

    private var topScorersData : List<TopScorers>? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TopScorersViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.top_scorer_row_template, parent, false)
        return TopScorersViewHolder(view)
    }

    override fun getItemCount(): Int {

        if(topScorersData == null){
            return 0
        }
        return topScorersData!!.size
    }

    override fun onBindViewHolder(holder: TopScorersViewHolder, position: Int) {
        holder.bindViews(topScorersData!![position])
    }

    fun updateTOpScorersData(topScorersDataList : List<TopScorers>){
        topScorersData = topScorersDataList
        notifyDataSetChanged()
    }

    inner class TopScorersViewHolder(itemView : View) : RecyclerView.ViewHolder(itemView){
        fun bindViews(topScorersObject: TopScorers) {

            itemView.top_scorer_name.text = topScorersObject.player.name
            itemView.top_scorer_team.text = topScorersObject.team.name
            itemView.top_scorer_goals.text = topScorersObject.goals.toString()
        }

    }

}